docs/core/extensions/caching.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Caching in .NET
+description: Learn how to use various in-memory and distributed caching mechanisms in .NET.
+author: IEvangelist
+ms.author: dapine
+ms.date: 07/20/2021
+---
+
+# Caching in .NET
+
+In this article, you'll learn about various caching mechanisms. Caching is the act of storing data in an intermediate-layer, making subsequent data retrievals faster. Conceptually, caching is a performance optimization strategy and design consideration. Caching can significantly improve app performance by making infrequently changing (or expensive to retrieve) data more readily available. This article introduces the two primary types of caching, and provides sample source code for both:
+
+- [`Microsoft.Extensions.Caching.Memory`](/dotnet/api/microsoft.extensions.caching.memory)
+- [`Microsoft.Extensions.Caching.Distributed`](/dotnet/api/microsoft.extensions.caching.distributed)
+
+> [!IMPORTANT]
+> There are two `MemoryCache` classes within .NET, one in the `System.Runtime.Caching` namespace and the other in the `Microsoft.Extensions.Caching` namespace:
+>
+> - <xref:System.Runtime.Caching.MemoryCache?displayProperty=fullName>
+> - <xref:Microsoft.Extensions.Caching.Memory.MemoryCache?displayProperty=fullName>
+>
+> While this article focuses on caching, it doesn't include the [`System.Runtime.Caching`](https://www.nuget.org/packages/System.Runtime.Caching) NuGet package. All references to `MemoryCache` are within the `Microsoft.Extensions.Caching` namespace.
```

Comment:
The `System.Runtime.MemoryCache` is still used, but introduces "static cling" and cannot be easily used with unit testing, whereas the `IMemoryCache` is literally registered as a service with DI. It can be mocked, or it's `ISystemClock` can be mocked. I would that for Puppet, your choice is perfectly valid.